### PR TITLE
Split submissions into 2 files depending on stage (nominee vs DPG) field value

### PR DESCRIPTION
**Pull request by nathanbaleeta**
Added component for `stage` (nominee vs DPG) to help manage submission logic.

### CODE REVIEW

**Comment by nathanbaleeta** [comment]
@lacabra the bug can be found in commit 06c983b. Thanks.

File: pages/api/openPR.js [hunk 1]
```diff
@@ -15,40 +15,199 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
...
+// Order fields e.g organizations
+function orderFields(values) {
+  let name, website, org_type;
+  // Order all entries for organizations
+  for (let i = 0; i < values.organizations.length; i++) {
+    name = values.organizations[i].name;
+    website = values.organizations[i].website;
+    org_type = values.organizations[i].org_type;
+
+    values.organizations[i] = {
```
Comment (lacabra) [inline pages/api/openPR.js:69]: You need to account for `organizations.contact_name` and `organizations.contact_email`

File: pages/api/openPR.js [hunk 2]
```diff
@@ -15,40 +15,199 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
...
+      name: name,
+      website: website,
+      org_type: org_type,
+    };
+  }
+  return values;
+}
+
+// Return multiple submission files including corresponding filepaths
+function getSubmissionFiles(values, myJSON) {
```
Comment (lacabra) [inline pages/api/openPR.js:79]: Please use more descriptive variables, instead of `myJSON` consider `nomineeJSON`

File: pages/api/openPR.js [hunk 3]
```diff
@@ -15,40 +15,199 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
...
+
+  // Add nominee submission to nominee directory
+  nomineePath = "nominees/" + `${name}`;
+
+  // Add DPG submission to screening directory
+  dpgPath = "screening/" + `${name}`;
+
+  stage === "nominee"
+    ? Object.entries(myJSON).forEach(
+        ([key, value]) =>
```
Comment (lacabra) [inline pages/api/openPR.js:116]: Why do you unpack `[key, value]` if they are never used?

File: pages/api/openPR.js [hunk 4]
```diff
@@ -15,40 +15,199 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
...
+          (files = {
+            [nomineePath]: {
+              content: myJSON,
+              encoding: "utf-8",
+            },
+          })
+      )
+    : Object.entries(myJSON).forEach(
+        ([key, value]) =>
```
Comment (lacabra) [inline pages/api/openPR.js:125]: Why do you unpack `[key, value]` if they are never used?

File: pages/api/openPR.js [hunk 5]
```diff
@@ -15,40 +15,199 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
...
+      })
+  );
+
+  // Order nominee fields in the correct order e.g organizations
+  sortedSubmission = orderFields(sortedSubmission);
+  return sortedSubmission;
+}
+
+// DPG candidate processing before opening pull request
+function dpgSubmission(values, sortedSubmission) {
```
Comment (lacabra) [inline pages/api/openPR.js:167]: How is this function different from `nomineeSubmission` ?

File: pages/api/openPR.js [hunk 6]
```diff
@@ -15,40 +15,199 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
...
 
-    // Return project json file aligned with naming convention (includes removing accents)
-    let name =
-      values.name
-        .normalize("NFD")
-        .toLowerCase()
-        .replace(/[\u0300-\u036f]/g, "")
-        .replace(/ /g, "-") + ".json";
+    // Convert JavaScript submission sorted object into JSON string and add newline at EOF
+    myJSON = JSON.stringify(sortedSubmission, null, 2).concat("\n");
```
Comment (lacabra) [inline pages/api/openPR.js:210]: Please use more descriptive variables, instead of `myJSON` consider `nomineeJSON`

File: pages/api/openPR.js [hunk 7]
```diff
@@ -15,40 +15,199 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
...
+      );
+
+  return files;
+}
+
+// Nominee processing before opening pull request
+function nomineeSubmission(values, sortedSubmission) {
+  // Sort entries by iterating through object and unpacking entries
+  Object.entries(values).forEach(
+    ([key, value]) =>
```
Comment (lacabra) [inline pages/api/openPR.js:145]: Why do you unpack [key, value] if they are never used?

**Review by lacabra — CHANGES_REQUESTED** [review]
Thanks @nathanbaleeta, this PR looks very good overall. Requested some minor adjustments.

File: pages/api/openPR.js (hunk 1 above)
Comment (nathanbaleeta) [reply pages/api/openPR.js:71]: Resolved in commit 8e456f7.

File: pages/api/openPR.js (hunk 3 above)
Comment (nathanbaleeta) [reply pages/api/openPR.js]: Fixed in commit 7703d7e.

File: pages/api/openPR.js (hunk 4 above)
Comment (nathanbaleeta) [reply pages/api/openPR.js]: Fixed in commit 7703d7e.

File: pages/api/openPR.js (hunk 7 above)
Comment (nathanbaleeta) [reply pages/api/openPR.js]: Fixed in commit 7703d7e.

File: pages/api/openPR.js (hunk 2 above)
Comment (nathanbaleeta) [reply pages/api/openPR.js]: Resolved in commit b645ff6.

File: pages/api/openPR.js (hunk 6 above)
Comment (nathanbaleeta) [reply pages/api/openPR.js]: Resolved in commit b645ff6.

File: pages/api/openPR.js (hunk 5 above)
Comment (nathanbaleeta) [reply pages/api/openPR.js]: Resolved in commit 6151335.

File: pages/api/openPR.js [hunk 8]
```diff
@@ -15,40 +15,178 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
...
-        evidenceText: values[evidenceText],
-      };
+    // Parse SDG information
+    values = getSDGRelevanceInfo(values, sdgNumber, evidenceText);
 
-      delete values[evidenceText];
-    }
+    // Verify submission stage(nominee/ DPG) and channel to nomination or DPG review processing
+    sortedSubmission =
+      values.stage === "nominee"
```
Comment (lacabra) [inline pages/api/openPR.js:184]: This ternary operator is now irrelevant if both conditions lead to the same result. This should be thus rewritten as:
`sortedSubmission = nomineeSubmission(values, sortedSubmission)`. Shouldn't it?

File: pages/api/openPR.js [hunk 9]
```diff
@@ -15,40 +15,178 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
...
+  name = parseProjectName(values);
+
+  // Add nominee submission to nominee directory
+  nomineePath = "nominees/" + `${name}`;
+
+  // Add DPG submission to screening directory
+  dpgPath = "screening/" + `${name}`;
+
+  stage === "nominee"
+    ? Object.entries(nomineeJSON).forEach(
```
Comment (lacabra) [inline pages/api/openPR.js:119]: I am still confused by this iteration of Object. This code repeats the same assignation of content to "files" for each entry in the object. This does not make any sense. Shouldn't be this rewritten as:
```
files = {
            [nomineePath]: {
              content: nomineeJSON,
              encoding: "utf-8",
            },
          }
```

File: pages/api/openPR.js [hunk 10]
```diff
@@ -15,40 +15,178 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
...
+        () =>
+          (files = {
+            [nomineePath]: {
+              content: nomineeJSON,
+              encoding: "utf-8",
+            },
+          })
+      )
+    : Object.entries(nomineeJSON).forEach(
```
Comment (lacabra) [inline pages/api/openPR.js:128]: Same as above, just assign value to `files` without `forEach()`

File: pages/api/openPR.js [hunk 11]
```diff
@@ -15,40 +15,178 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
...
+          })
+      );
+
+  return files;
+}
+
+// Nominee processing before opening pull request
+function nomineeSubmission(values, sortedSubmission) {
+  // Sort entries by iterating through object and unpacking entries
+  Object.entries(values).forEach(
```
Comment (lacabra) [inline pages/api/openPR.js:148]: Same observation as above, iterating through the object is not needed.

**Review by lacabra — CHANGES_REQUESTED** [review]
Thanks @nathanbaleeta, additional request for changes based on your changes.

File: pages/api/openPR.js (hunk 8 above)
Comment (nathanbaleeta) [reply pages/api/openPR.js]: Fixed in commit 9509763.

File: pages/api/openPR.js (hunk 10 above)
Comment (nathanbaleeta) [reply pages/api/openPR.js]: Resolved in commit dd5cfcf.

File: pages/api/openPR.js (hunk 11 above)
Comment (nathanbaleeta) [reply pages/api/openPR.js]: Resolved in commit dd5cfcf.

File: pages/api/openPR.js (hunk 9 above)
Comment (nathanbaleeta) [reply pages/api/openPR.js]: Resolved in commit dd5cfcf.

**Review by lacabra — APPROVED** [review]
Everything looks great now, many many thanks @nathanbaleeta 🙏

**Pull request merged**